### PR TITLE
UCLA custom handler

### PR DESCRIPTION
**Pull request by emord**
https://docs.google.com/document/d/1Hc7Cd53AcQJ5o1WybukAHdaUnbEtwDgMbXB0waXBUpc/edit

@gcapalbo buddy: @benrudolph

### CODE REVIEW

File: custom/ucla/tests.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
+from collections import namedtuple
+import uuid
+from django.test import TestCase
+
+from corehq.apps.fixtures.models import (
+    FixtureDataType, FixtureTypeField, FixtureDataItem, FieldList, FixtureItemField
+)
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
```
Comment (dimagimon) [inline custom/ucla/tests.py]: F401 'CaseAccessors' imported but unused

File: custom/ucla/api.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        lambda m: m.fields_without_attributes['sequence'] == current_message_seq_num,
+        custom_messages
+    )
+
+    if len(custom_messages) != 1:
+        if not custom_messages:
+            message = "No message for risk {}, seq {} in domain {}"
+        else:
+            message = "Multiple messages for risk {}, seq {} in domain {}"
+        message = message.format(risk_profile, current_message_seq_num, domain)
```
Comment (benrudolph) [inline custom/ucla/api.py]: nit: if risk profile can be unicode this could through an error

**Review by benrudolph — APPROVED** [review]
👍 one question about the `notify_exception`. i don't have too much context, but is that a kind of error that we can do something about? from this is looks like it should almost be a message sent back to the caller of the api

File: custom/ucla/tests.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+class UCLACustomHandler(TestCase):
+    domain = uuid.uuid4().hex
+    case_type = 'ucla-reminder'
+    fixture_name = 'message_bank'
+
+    def _reminder(self):
+        return Reminder(
+            domain=self.domain,
+            schedule_iteration_num=0,
```
Comment (gcapalbo) [inline custom/ucla/tests.py]: we should make this be 1 by default (the iteration num starts counting at 1)

File: custom/ucla/tests.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            current_event_sequence_num=0,
+        )
+
+    def _handler(self, num_events=0):
```
Comment (gcapalbo) [inline custom/ucla/tests.py]: i think it would be good for all the tests to use num_events=1 for this

File: custom/ucla/tests.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    def test_no_relevant_message_invalid_risk(self):
+        self._setup_fixture_type()
+        with create_test_case(self.domain, self.case_type, 'test-case', {'risk_profile': 'risk2'}) as case:
+            self.assertIsNone(ucla_message_bank_content(self._reminder(), self._handler(), case))
+
+    @run_with_all_backends
+    def test_no_relevant_message_invalid_seq_num(self):
+        self._setup_fixture_type()
+        with create_test_case(self.domain, self.case_type, 'test-case', {'risk_profile': 'risk1'}) as case:
+            self.assertIsNone(ucla_message_bank_content(self._reminder(), self._handler(num_events=1), case))
```
Comment (gcapalbo) [inline custom/ucla/tests.py]: we could use a reminder with schedule_iteration_num=2 for this test, keeping num_events=1 on the handler

File: custom/ucla/api.py (hunk 2 above)
Comment (gcapalbo) [inline custom/ucla/api.py]: i think to be safe we could just make all the notify messages be unicode by prepending `u` to them

**Comment by gcapalbo** [comment]
looks great @emord, just a few comments

@benrudolph regarding the notify_exception, yeah maybe that's not the best fit. these errors can happen at reminder runtime, so i was thinking there should be some way to alert us of config issues in the project so we can fix them. maybe a better thing to do would be to send an email to all "dimagi.com" web users in the project rather than notify_exception.

File: custom/ucla/tests.py [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    def test_no_relevant_message_invalid_risk(self):
+        self._setup_fixture_type()
+        with create_test_case(self.domain_name, self.case_type, 'test-case', {'risk_profile': 'risk2'}) as case:
+            self.assertRaises(AssertionError, ucla_message_bank_content, self._reminder(), self._handler(), case)
+
+    @run_with_all_backends
+    def test_no_relevant_message_invalid_seq_num(self):
+        self._setup_fixture_type()
+        with create_test_case(self.domain_name, self.case_type, 'test-case', {'risk_profile': 'risk1'}) as case:
+            self.assertRaises(AssertionError, ucla_message_bank_content, self._reminder(iteration_num=2), self._handler(), case)
```
Comment (dimagimon) [inline custom/ucla/tests.py]: E501 line too long (128 > 115 characters)

**Comment by emord** [comment]
Updated to include all the feedback. I went with soft assert to email all of the dimagi people on the project and if there aren't any, then fall back to notifying admins

**Review by gcapalbo — APPROVED** [review]

**Comment by gcapalbo** [comment]
looks great, just waiting on tests

**Pull request merged**